src/app/webbrowser/BrowserTab.qml

```diff
@@ -190,40 +184,36 @@ FocusScope {
 
             internal.hiding = true
             webview.grabToImage(function(result) {
-                if (!internal.hiding) {
-                    return
-                }
-                internal.hiding = false
                 visible = false
+                preview = result.url
+                previewCache = result
+            },previewSize);
 
+            //save previews to disk for newtabpage and tab during grabbing
+            webview.grabToImage(function(result) {
+                internal.hiding = false
                 PreviewManager.saveToDisk(result, url)
-            })
+            },previewThumbnailSize);
```

Comment:
I'm unsure where it comes from, but the following warning is thrown when tabs become visible:

```
Ignoring sourceSize request for image url that came from grabToImage. Use the targetSize parameter of the grabToImage() function instead.
```

Reply:
This is coming from depths of QT when HIDPI is used. Probably QT bug, there is chance that it will fix the new QT. I am using targetSize parameter, not sure why this warning.